### PR TITLE
Added Tests for Moon Phase Script

### PR DESCRIPTION
**Pull request by CoDinAIy**
# Description
- Tests for moon phase extract script!
- Modified the script itself to return error message instead of printing then returning `None`

Closes: #104 

## Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

# Mentions
@emily-james
@CoDinAIy
@rosie-13
@mercuridi

### CODE REVIEW

File: test/test_moon_phase_extract.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import pytest
+from unittest.mock import MagicMock, patch
+from src.moon_phase_extract import get_moon_phase_image
+from src.astronomy_utils import make_request_headers
+
+
+@patch('requests.post')
```
Comment (rosie-13) [inline test/test_moon_phase_extract.py:7]: Praise: Nice use of patching requests

File: test/test_moon_phase_extract.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    mock_post_response.json.return_value = {
+        'data': {
+            'imageUrl': 'https://fakeurl.com/moon.jpg'
+        }
+    }
+    mock_post.return_value = mock_post_response
+
+    mock_get_response = MagicMock()
+    mock_get_response.status_code = 200
+    mock_get_response.content = b'fake_image_data'
```
Comment (rosie-13) [inline test/test_moon_phase_extract.py:21]: Question: what does the `b` do here?

File: test/test_moon_phase_extract.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import pytest
+from unittest.mock import MagicMock, patch
+from src.moon_phase_extract import get_moon_phase_image
+from src.astronomy_utils import make_request_headers
+
+
+@patch('requests.post')
+@patch('requests.get')
+def test_moon_phase_image_gotten(mock_get, mock_post):
```
Comment (rosie-13) [inline test/test_moon_phase_extract.py:9]: Nitpick: Could this test be rephrased to something like `test_moon_phase_image_successfully_retrieved` for clarity?

File: test/test_moon_phase_extract.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    with patch("requests.get") as mock_get:
+        mock_get.return_value.content = b'image_data'
+        moon_phase_url = "https://api.astronomyapi.com/api/v2/studio/moon-phase"
+        get_moon_phase_image(0.05, 0.14, '2025-03-02',
+                             {}, moon_phase_url)
+        mock_post.assert_called_once()
+
+
+def test_moon_phase_image_invalid():
```
Comment (rosie-13) [inline test/test_moon_phase_extract.py:50]: Praise: Nice checking of edge cases

File: test/test_moon_phase_extract.py (hunk 2 above)
Comment (CoDinAIy) [reply test/test_moon_phase_extract.py:21]: It presents the `content` as a bytes literal. Since the function returns image data in bytes, this mocks the response as if it was a real image data in bytes

Comment (rosie-13) [reply test/test_moon_phase_extract.py:21]: Ah makes sense, thanks!

**Review by rosie-13 — APPROVED** [review]
